markovflow/models/spatio_temporal_variational.py

```diff
@@ -270,12 +341,256 @@ def posterior(self) -> PosteriorProcess:
+        * p(y | f) - the likelihood
+        * p(.) the prior distribution
+        * q(.) the variational distribution
+
+    This can be seen as the spatial extension of markovflow's SparseCVIGaussianProcess
+    for temporal (only) Gaussian Processes.
+    The inducing variables u(x,t) are now space and time dependent.
+
+    for a fixed set of space points zₛ
+    p(x(zₛ, .)) is a continuous time process of state dimension Mₛd
+    for a fixed time slice t, p(x(.,t)) ~ GP(0, kₛ)
+
+    The following conditional independence holds:
+    p(x(s,t) | x(zₛ, .)) = p(x(s,t) | s(zₛ, t)), i.e.,
+    prediction at a new point at time t given x(zₛ, .) only depends on s(zₛ, t)
+
+    This builds a spatially sparse process as
+    q(x(.,.)) = q(x(zₛ, .)) p(x(.,.) |x(zₛ, .)),
+    where the multi-output temporal process q(x(zₛ, .)) is also sparse
+    q(x(zₛ, .)) = q(x(zₛ, zₜ)) p(x(zₛ,.) |x(zₛ,  zₜ))
+
+    the marginal q(x(zₛ, zₜ)) is parameterized as the product
+    q(x(zₛ, zₜ)) = p(x(zₛ, zₜ)) t(x(zₛ, zₜ))
+    where p(x(zₛ, zₜ)) is a state space model and t(x(zₛ, zₜ)) are sites.
+    """
+
+    def __init__(
+        self,
+        inducing_space,
+        inducing_time,
+        kernel_space: gpfk.Kernel,
+        kernel_time: mfk.SDEKernel,
+        likelihood: gpflow.likelihoods.Likelihood,
+        mean_function: Optional[MeanFunction] = None,
+        num_data=None,
+        learning_rate=0.1,
+    ) -> None:
+        """
+        :param inducing_space: inducing space points [Ms, D]
+        :param inducing_time: inducing time points [Mt,]
+        :param kernel_space: Gpflow space kernel
+        :param kernel_time: Markovflow time kernel
+        :param likelihood: a likelihood object
+        :param mean_function: The mean function for the GP. Defaults to no mean function.
+        :param num_data: The total number of observations.
+            (relevant when feeding in external minibatches).
+        :param learning_rate: the learning rate.
+        """
+
+        super().__init__(inducing_space, kernel_space, kernel_time, likelihood, mean_function)
+
+        self.num_data = num_data
+        self._inducing_time = inducing_time
+        self.num_inducing_time = inducing_time.shape[0]
+
+        self.learning_rate = learning_rate
+        # initialize sites
+        num_inducing = inducing_time.shape[0]
+        state_dim = self._kernel.state_dim
+        zeros1 = tf.zeros((num_inducing + 1, 2 * state_dim), dtype=default_float())
+        zeros2 = tf.zeros((num_inducing + 1, 2 * state_dim, 2 * state_dim), dtype=default_float())
+        self.nat1 = Parameter(zeros1)
+        self.nat2 = Parameter(zeros2)
+
+    @property
+    def posterior(self) -> ConditionalProcess:
+        """ Posterior object to predict outside of the training time points """
+        return ConditionalProcess(
+            posterior_dist=self.dist_q,
+            kernel=self.kernel,
+            conditioning_time_points=self._inducing_time,
+        )
```

Comment:
Would it make sense to define this as `self._posterior` in `__init__` as `SpatioTemporalSparseVariational` does and maybe also move the getter to the base class?